messenger.go

```diff
@@ -317,6 +318,55 @@ func NewMessenger(
 	return messenger, nil
 }
 
+// Init analyzes chats and contacts in order to setup filters
+// which are responsible for retrieving messages.
+func (m *Messenger) Init() error {
+	logger := m.logger.With(zap.String("site", "Init"))
+
+	var (
+		publicChatIDs []string
+		publicKeys    []*ecdsa.PublicKey
+	)
+
+	// Get chat IDs and public keys from the existing chats.
+	// TODO: Get only active chats by the query.
+	chats, err := m.Chats()
+	if err != nil {
+		return err
+	}
+	for _, chat := range chats {
+		if !chat.Active {
+			continue
+		}
+		if chat.ChatType == ChatTypePublic {
+			publicChatIDs = append(publicChatIDs, chat.ID)
+		} else if chat.ChatType == ChatTypeOneToOne {
+			publicKeys = append(publicKeys, chat.PublicKey)
+		}
```

Comment:
Forgot to mention sorry, we also need to create a filter for each group chat member, `ChatMember.ID` is the hex encoded public key

Reply:
Added.

Reply:
looks good, thanks, just a note that you might have duplicates, but if we are fine with that and they are handled properly then no probs

Reply:
It is. If a filter of the same type for a given public key exists, it will be returned early without creating a duplicate.